addons/controllers/clusterbootstrap_controller.go

```diff
@@ -1342,14 +1396,31 @@ func (r *ClusterBootstrapReconciler) watchProvider(providerRef *corev1.TypedLoca
 // Returns:
 // - string: The secret name which references to the Secret CR on mgmt cluster under a particular cluster namespace.
 // - error: whether there is error when getting the secret name.
-func (r *ClusterBootstrapReconciler) GetDataValueSecretNameFromBootstrapPackage(cbPkg *runtanzuv1alpha3.ClusterBootstrapPackage, clusterNamespace string) (string, error) {
+func (r *ClusterBootstrapReconciler) GetDataValueSecretNameFromBootstrapPackage(cbPkg *runtanzuv1alpha3.ClusterBootstrapPackage, cluster *clusterapiv1beta1.Cluster) (string, error) {
 	// When valuesFrom is nil, we interpret it as no data values are needed for the package installation.
 	if cbPkg.ValuesFrom == nil {
 		r.Log.Info(fmt.Sprintf("no data values are provided to the ClusterBootstrapPackage.ValuesFrom field. ClusterBootstrapPackage.RefName: %s", cbPkg.RefName))
 		return "", nil
 	}
 
-	// TODO: Handle inline valueFrom. https://github.com/vmware-tanzu/tanzu-framework/issues/1694
+	if cbPkg.ValuesFrom.Inline != nil {
+		packageRefName, _, err := util.GetPackageMetadata(r.context, r.aggregatedAPIResourcesClient, cbPkg.RefName, cluster.Namespace)
```

Comment:
suggestion(non-blocking): since we are using the live client `aggregatedAPIResourcesClient`, that means each call will talk to apiserver directly. And there are 6 places in clusterbootstrap_controller invokes it. We might want to file a GH issue to have it improved: reuse the result of util.GetPackageMetadata() as much as we could.

Reply:
this can be done in follow up PR